public/js/use.js

```diff
@@ -280,6 +290,84 @@ function setupUse(params) {
     }
   }
 
+  // need to add on resize event listener for faces
+  // need to offset and position itself and scale properly with physical image location
+  // need to calculate ratio of image
+  // get ratio
+  // get starting image position
+  // get transformed positions of face
+  //  = ratio * original positions + offset
+  function renderEntities(results) {
+    // eslint-disable-next-line camelcase
+    var imageBoxes_template;
+    if (results.images[0].faces) {
+      // eslint-disable-next-line camelcase
+      imageBoxes_template = imageBoxesTemplate.innerHTML;
+      var faceLocations = results.images[0].faces.map(function(face) {
+        return transformBoxLocations(face.face_location, document.querySelector('.use--output-image'));
+      });
+
+      console.log(faceLocations);
+
+      $boxes.append(_.template(imageBoxes_template, {
+        items: faceLocations
+      }));
+    }
+
+    if (results.images[0].words) {
+      // eslint-disable-next-line camelcase
+      imageBoxes_template = imageBoxesTemplate.innerHTML;
+      var locations = results.images[0].words.map(function(word) {
+        return transformBoxLocations(word.location, document.querySelector('.use--output-image'));
+      });
+
+      console.log(locations);
+
+      $boxes.append(_.template(imageBoxes_template, {
+        items: locations
+      }));
+    }
+
+    console.log(results.images[0].words);
+  }
+
+  function transformBoxLocations(faceLocation, image) {
+    var newFaceLocation = faceLocation;
+    var ratio = image.getBoundingClientRect().width / image.naturalWidth;
+    var coordinates = getCoords(image);
+    console.log(image.getBoundingClientRect());
+    newFaceLocation = {
+      width: faceLocation.width * ratio,
+      height: faceLocation.height * ratio,
+      top: coordinates.top + faceLocation.top * ratio,
+      left: coordinates.left + faceLocation.left * ratio
+    };
+    console.log(image.getBoundingClientRect(), ratio, image.getBoundingClientRect().width, faceLocation.width);
+    return newFaceLocation;
+  }
+
+  /*
+   * Solution found here:
+   * http://stackoverflow.com/questions/5598743/finding-elements-position-relative-to-the-document#answer-26230989
+   */
+  function getCoords(elem) { // crossbrowser version
+    var box = elem.getBoundingClientRect();
```

Comment:
Is this any different from `jQuery(elm).offset();` ?  http://api.jquery.com/offset/